src/test/java/com/microsoft/azure/spring/data/cosmosdb/domain/inheritance/Square.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.data.cosmosdb.domain.inheritance;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.data.annotation.Id;
+
+@Data
+@NoArgsConstructor
+public class Square extends Shape {
```

Comment:
If this is a square, the area could be calculated by length * length rather than passing the area as one extra parameter.

Reply:
This is used to demonstrate inheritance but not real usage. I thought this a bit but think it is OK for inheritance domain purpose.